api/v1alpha1/nodehealthcheck_webhook.go

```diff
@@ -167,18 +178,37 @@ func (nhc *NodeHealthCheck) validateEscalatingRemediationsTimeout() error {
 	return nil
 }
 
-func (nhc *NodeHealthCheck) validateEscalatingRemediationsUniqueRemediator() error {
-	// this is a workaround until we designed a way to support multiple remediators of same kind
+func (v *customValidator) validateEscalatingRemediationsUniqueRemediator(ctx context.Context, nhc *NodeHealthCheck) error {
 	remediators := make(map[string]struct{}, len(nhc.Spec.EscalatingRemediations))
 	for _, rem := range nhc.Spec.EscalatingRemediations {
-		if _, exists := remediators[rem.RemediationTemplate.Kind]; exists {
+		kind := rem.RemediationTemplate.Kind
+		if _, exists := remediators[kind]; exists && !v.isMultipleTemplatesSupported(ctx, rem.RemediationTemplate) {
 			return fmt.Errorf("%s: duplicate template kind: %v", uniqueRemediatorError, rem.RemediationTemplate.Kind)
 		}
 		remediators[rem.RemediationTemplate.Kind] = struct{}{}
 	}
 	return nil
 }
 
+func (v *customValidator) isMultipleTemplatesSupported(ctx context.Context, nhcExpectedTemplate corev1.ObjectReference) bool {
+	templateCRBase := &unstructured.Unstructured{}
+	templateCRBase.SetGroupVersionKind(nhcExpectedTemplate.GroupVersionKind())
+	templateList := &unstructured.UnstructuredList{Object: templateCRBase.Object}
+
+	if err := v.Client.List(ctx, templateList); err != nil || len(templateList.Items) == 0 {
```

Comment:
I think so, I think allowing mixed templates (templates of the same kind that some are annotated with support and some aren't)  is a rabbit hole we should probably avoid.
One way to get into this sort of situation is an upgrade scenario.

[here](https://github.com/medik8s/node-healthcheck-operator/blob/main/controllers/resources/templates.go#L49-L56) is one example of how this can cause an issue.

Assuming there are 2 templates of the same kind: Template B (with supported annotation) and Template A (without annotation)
In case template A precedes B, B will never be tested because A will always return a false indication that the remediation already occurred.